index.js

```diff
@@ -842,15 +842,27 @@ const installMaybe = () => {
           .then(saveCachedNpm)
           .then(saveCachedCypressBinary)
       })
-    })
+    }))
   })
 }
 
-installMaybe()
-  .then(buildAppMaybe)
-  .then(startServersMaybe)
-  .then(waitOnMaybe)
-  .then(runTests)
+// Wrap a function such that any output it produces ends up in a folded log
+// section with a title of $name
+// The fn argument doesn't have to return a promise, but if it does, the
+// folding section will end when that promise is settled.
+const logGroup = (name, fn) => {
+  core.startGroup(name);
+  return (resolved) => {
+    return Promise.resolve(fn(resolved))
+      .finally(core.endGroup)
+   }
+}
+
+logGroup("Restore cache and install", installMaybe)
```

Comment:
I think there's a bug here.

This should be:

```suggestion
logGroup("Restore cache and install", installMaybe)()
```